Documentation/benchmarks/etcd-3-watch-memory-benchmark.md

```diff
@@ -0,0 +1,72 @@
+# Watch Memory Usage Benchmark
+
+*NOTE*: The watch features are still being actively developed and we expect the memory usage to change over time. However we do expect that memory usage should not be significantly increased from the figures below.
+
+A primary goal of etcd is to support large numbers of watching. etcd aims to support as many as O(10k) clients, O(100K) watch streams (O(10) streams per client) and O(10M) total watchings (O(100) watching per stream). Thus the goal is to minimize the memory consumption per watching, since it will dominate the overall consumption.
+
+Three related components of etcd watch consume physical memory: grpc.Conn, watch stream, watching. grpc.Conn maintains the actual TCP connection and other gRPC connection state. Each grpc.Conn consumes O(10kb) of memory.  Each grpc.Conn might have multiple watch streams attached. 
```

Comment:
"Three related components of etcd watch consume physical memory: each `grpc.Conn`, each watch stream, and each instance of the watching activity."